scalafix/build.sbt

```diff
@@ -0,0 +1,96 @@
+lazy val V = _root_.scalafix.sbt.BuildInfo
+
+lazy val rulesCrossVersions = Seq(V.scala213, V.scala212)
+lazy val scala3Version      = "3.5.0"
+
+inThisBuild(
+  List(
+    organization := "typelevel",
+    homepage     := Some(url("https://github.com/typelevel/weaver-test")),
+    licenses := List(
+      "Apache-2.0" -> url("http://www.apache.org/licenses/LICENSE-2.0")
+    ),
+    semanticdbEnabled := true,
+    semanticdbVersion := scalafixSemanticdb.revision
+  )
+)
+
+lazy val `weaver-test` = (project in file("."))
+  .aggregate(
+    rules.projectRefs ++
+      input.projectRefs ++
+      output.projectRefs ++
+      tests.projectRefs: _*
+  )
+  .settings(
+    publish / skip := true
+  )
+
+lazy val rules = projectMatrix
+  .settings(
+    moduleName := "scalafix",
+    libraryDependencies += "ch.epfl.scala" %% "scalafix-core" % V.scalafixVersion
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(rulesCrossVersions)
+
+lazy val input = projectMatrix
+  .settings(
+    publish / skip                               := true,
+    libraryDependencies += "com.disneystreaming" %% "weaver-cats" % "0.8.3"
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
+
+lazy val output = projectMatrix
+  .settings(
+    publish / skip                               := true,
+    libraryDependencies += "com.disneystreaming" %% "weaver-cats" % "0.8.3"
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
+
+lazy val testsAggregate = Project("tests", file("target/testsAggregate"))
+  .aggregate(tests.projectRefs: _*)
+  .settings(
+    publish / skip := true
+  )
+
+lazy val tests = projectMatrix
+  .settings(
+    publish / skip := true,
+    scalafixTestkitOutputSourceDirectories :=
+      TargetAxis
+        .resolve(output, Compile / unmanagedSourceDirectories)
+        .value,
+    scalafixTestkitInputSourceDirectories :=
+      TargetAxis
+        .resolve(input, Compile / unmanagedSourceDirectories)
+        .value,
+    scalafixTestkitInputClasspath :=
+      TargetAxis.resolve(input, Compile / fullClasspath).value,
+    scalafixTestkitInputScalacOptions :=
+      TargetAxis.resolve(input, Compile / scalacOptions).value,
+    scalafixTestkitInputScalaVersion :=
+      TargetAxis.resolve(input, Compile / scalaVersion).value
+  )
+  .defaultAxes(
+    rulesCrossVersions.map(VirtualAxis.scalaABIVersion) :+ VirtualAxis.jvm: _*
+  )
+  .jvmPlatform(
+    scalaVersions = Seq(V.scala212),
+    axisValues = Seq(TargetAxis(scala3Version)),
+    settings = Seq()
+  )
```

Comment:
Can you explain the need for the custom `TargetAxis` please? (also I suppose it has something to do with why this is mixing 2.12 and 3.x)